File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/DungeonPartyFinderRoomContentMessage.scala





















// Generated on 05/03/2013 10:34:46
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait DungeonPartyFinderRoomContentMessage extends MessageDeserializer {
  val messageId = 6247

  protected trait Message extends BaseMessage with Serializable {
    val messageId = DungeonPartyFinderRoomContentMessage.messageId

    val dungeonId: Short
    val players: Seq[DungeonPartyFinderPlayer.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeShort(dungeonId)
      buf.writeUShort(players.length)
      for (entry <- players) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(dungeonId: Short, players: Seq[DungeonPartyFinderPlayer.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val dungeonId = buf.readShort()
    val players = {
      val builder = collection.immutable.Seq.newBuilder[DungeonPartyFinderPlayer.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += DungeonPartyFinderPlayer.deserialize(buf)
      }
      builder.result()
    }

    create(dungeonId, players)
  }

}

object DungeonPartyFinderRoomContentMessage extends DungeonPartyFinderRoomContentMessage {
  case class MessageImpl(dungeonId: Short, players: Seq[DungeonPartyFinderPlayer.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(dungeonId: Short, players: Seq[DungeonPartyFinderPlayer.TypeImpl]) = MessageImpl(dungeonId, players)

  def apply(dungeonId: Short, players: Seq[DungeonPartyFinderPlayer.TypeImpl]) = create(dungeonId, players)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/DungeonPartyFinderRoomContentMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class DungeonPartyFinderRoomContentMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6247;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short dungeonId;
    public DungeonPartyFinderPlayer[] players;
    
    public DungeonPartyFinderRoomContentMessage() { }
    
    public DungeonPartyFinderRoomContentMessage(short dungeonId, DungeonPartyFinderPlayer[] players) {
        this.dungeonId = dungeonId;
        this.players = players;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(dungeonId);
        buf.writeUShort(players.length);
        for (DungeonPartyFinderPlayer entry : players) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        dungeonId = buf.readShort();
        if (dungeonId < 0)
            throw new RuntimeException("Forbidden value on dungeonId = " + dungeonId + ", it doesn't respect the following condition : dungeonId < 0");
        int limit = buf.readUShort();
        players = new DungeonPartyFinderPlayer[limit];
        for (int i = 0; i < limit; i++) {
            players[i] = new DungeonPartyFinderPlayer();
            players[i].deserialize(buf);
        }
    }
    
}
